CUMULUS-2394 -- Update granule/pdr writes to not allow old workflows to overwrite new records (#2123)
nasa/cumulus: CHANGELOG.md

### Added

- `tf-modules/cumulus` module now supports a `cmr_custom_host` variable that can
  be used to set to an arbitray  host for making CMR requests (e.g.
  `https://custom-cmr-host.com`).
- Added `buckets` variable to `tf-modules/archive`

### Fixed

- **CUMULUS-2394**
  - Updated PDR and Granule writes to check the step function start_time against
      the `createdAt` fild  for each record to ensure old execution records do not
      overwrite newer executions

### Changed

> - `<prefix>-lambda-api-gateway` IAM role used by API Gateway Lambda now supports

Author: Updates below are using re-wrap on the CHANGELOG

### Added

- `tf-modules/cumulus` module now supports a `cmr_custom_host` variable that can
  be used to set to an arbitray  host for making CMR requests (e.g.
  `https://custom-cmr-host.com`).
- Added `buckets` variable to `tf-modules/archive`
- **CUMULUS-2328**
  - Added `tf-modules/s3_credentials` module which contains resources to attach the `/s3-credentials` endpoint to an API gateway

### Fixed

- **CUMULUS-2394**

>   - Updated PDR and Granule writes to check the step function start_time against

Reviewer: ```suggestion
  - Updated PDR and Granule writes to check the step function `workflow_start_time` against
```

>       the `createdAt` fild  for each record to ensure old execution records do not

Reviewer: ```suggestion
      the `createdAt` field  for each record to ensure old execution records do not
```

>       the `createdAt` field  for each record to ensure old execution records do not
>       overwrite newer executions

Reviewer: ```suggestion
      the `createdAt` field  for each record to ensure old records do not
      overwrite newer ones
```

Nit to make it clearer that we're not talking about execution records

Author: 👍🏻
